File: board-room/src/main/java/net/stzups/board/server/websocket/protocol/client/ClientMessageOpenDocument.java
package net.stzups.board.server.websocket.protocol.client;

public class ClientMessageOpenDocument extends ClientMessage {
    private long id;

    public ClientMessageOpenDocument(long id) {
        super(ClientMessageType.OPEN_DOCUMENT);
        this.id = id;
    }

    public long getId() {
        return id;
    }
}

File: scribbleshare-app/scribbleshare-frontend/src/scripts/protocol/client/messages/ClientMessageOpenDocument.ts
import ClientMessageType from "../ClientMessageType.js";
import ClientMessage from "../ClientMessage.js";
import ByteBuffer from "../../ByteBuffer.js";
import Document from "../../../Document.js";

export default class ClientMessageOpenDocument extends ClientMessage {
    id: bigint;

    constructor(document: Document) {
        super(ClientMessageType.OPEN_DOCUMENT);
        this.id = document.id;
    }

    serialize(byteBuffer: ByteBuffer) {
        super.serialize(byteBuffer);
        byteBuffer.writeBigInt64(this.id);
    }
}